### ISSUE TITLE
BSD-isms in stdlib

### ISSUE THREAD

**Issue by RomanHargrave** [scala-native/scala-native#88]
Linux user here.

I noticed that in order to get a working build of the demo on Linux, `__stderrp` has to be changed to `stderr` due to differences in GNU libc and BSD libc WRT to the two standard IO streams. On BSDs, `stderr` is apparently a preprocessor macro that emits  `__stderrp`. The scala-native stdlib currently assumes a BSD environment, whereas it should conditionally select `__stdXp` or `stdx` based on the target platform.

**Comment by densh** [comment]
Hi @RomanHargrave, that's temporary. We're going to have support for conditional compilation based on target platform soon enough (https://github.com/scala-native/scala-native/issues/58). I've just started working on it in https://github.com/scala-native/scala-native/commits/topic/cross-target. This is going to resolve everybody's problem with building it on linux.

I'm closing this issue as duplicate of the one I've linked before.
